### DISCUSSION TITLE
Is something like Blazegraph GAS searches feasible in the future?

### DISCUSSION THREAD

**Discussion by fils** [oxigraph/oxigraph#81] [General]
For background see: https://github.com/oxigraph/oxigraph/issues/79 

@Tpt Thanks for the detailed response.     Looking forward to learning more about RDF* and how it relates.  Your overview of these topics helped..  thanks for that.   

WRT to GAS, I mostly play with BFS with target vertices to help explore resource relations.  I've been hoping I could find some ways to do this in just core W3C SPARQL.  Perhaps with multiple queries and code logic, but just haven't had the time.    

For me, my main concerns about Blazegraph are more around what it's future looks like.   From what I know, AWS's Neptune is based on Blazegraph and I don't know how many people moved to AWS from the team.  I also don't know what the commitment back to the opensource blazegraph looks like.  Perhaps it is solid, but I can't yet convince myself of that.  

Performance wise I deal with rather small graphs (10's of millions, perhaps 100 or 200 million max).   So not large by graph standards.   My community is solidly in the RDF/SPARQL camp (I do a lot of work with geoinformatics groups funded by the National Science Foundation).   I've been looking around for alternative triplestores, so I am happy to find oxigraph and explore it.  

BTW, I fully agree with your full text search and RDF* priorities.  As soon as full text search is in I'll start using oxigraph for some my test environments.

**Comment by Tpt** [comment]
> For me, my main concerns about Blazegraph are more around what it's future looks like. From what I know, AWS's Neptune is based on Blazegraph and I don't know how many people moved to AWS from the team. I also don't know what the commitment back to the opensource blazegraph looks like. Perhaps it is solid, but I can't yet convince myself of that.

The full team moved to AWS and the Blazegraph development completely stopped. After the acquisition there just have been a "bug fix" release done with some patches written by Wikimedia Foundation people who use Blazegraph for Wikidata. As long as Wikidata uses Blazegraph, bug fix for the features used by Wikimedia will probably still going to be written but probably not much more will happen. 

10th and 100th of million is the scale where I hope Oxigraph will shine. I don't think it's there yet for complex queries. Hopefully in the future.

Thank you gain for being interested in Oxigraph.

**Reply by fils** [reply]
I was worried that was the case.   I have a small connection with some of the people behind WikiData (a few exchanges around some potential NSF proposals).  I should reach out to them and see if they can share current thinking about the state of Blaze and what if any exist strategy they are thinking with it.

**Reply by akuckartz** [reply]
A relevant Wikibase issue https://phabricator.wikimedia.org/T206560
